File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int left, int right) {
    int r=0,t;
    for(int n=left;n<right+1;n++) {
        t=0;
        for(int i=1;i<n+1;i++) {
            if(n%i==0)
                t+=1;
        }
        r+=(t%2==0)?n:-n;
    }
    return r;
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
class Solution {
    public int solution(int left, int right) {
        int rlt = 0, cnt = 0;
        for(int i=left;i<right+1;i++) {
            cnt = 0;
            for(int j=1;j<i+1;j++) {
                if(i%j==0)
                    cnt+=1;
            }
            rlt+=cnt%2==0?i:-i;
        }
        return rlt;
    }
}